Grpc Interop tests assembly load failure on Helix

The gRPC interop tests are currently disabled.

We saw failures after the latest dependency updates brought breaking Logging changes in the runtime: https://dev.azure.com/dnceng/public/_build/results?buildId=739743&view=results

The error suggests that the wrong version of the M.E.Logging.Console is being loaded at runtime:
```
[ERROR] Unhandled exception. System.TypeLoadException: Could not load type 'Microsoft.Extensions.Logging.Console.SimpleConsoleFormatterOptions' from assembly 'Microsoft.Extensions.Logging.Console, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
[ERROR]    at InteropTestsWebsite.Program.<>c.<CreateHostBuilder>b__1_0(ILoggingBuilder builder)
[ERROR]    at Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.<>c__DisplayClass5_0.<ConfigureLogging>b__1(ILoggingBuilder builder)
[ERROR]    at Microsoft.Extensions.DependencyInjection.LoggingServiceCollectionExtensions.AddLogging(IServiceCollection services, Action`1 configure)
[ERROR]    at Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.<>c__DisplayClass5_0.<ConfigureLogging>b__0(HostBuilderContext context, IServiceCollection collection)
[ERROR]    at Microsoft.Extensions.Hosting.HostBuilder.CreateServiceProvider()
[ERROR]    at Microsoft.Extensions.Hosting.HostBuilder.Build()
[ERROR]    at InteropTestsWebsite.Program.Main(String[] args) in F:\workspace\_work\1\s\src\Grpc\test\testassets\InteropWebsite\Program.cs:line 33
Writing minidump to file /home/helixbot/dotnetbuild/dumps/dotnet.31827.dmp
Written 93696000 bytes (22875 pages) to core file
Dump successfully written
```

I suspect that the helix environment didn't install the correct aspnetcore runtime and therefore the M.E.Logging.Console binary that was loaded is old.

--- a/src/Grpc/test/InteropTests/InteropTests.csproj
+++ b/src/Grpc/test/InteropTests/InteropTests.csproj
@@ -3,8 +3,6 @@
   <PropertyGroup>
     <ContainsFunctionalTestAssets>true</ContainsFunctionalTestAssets>
     <TargetFramework>$(DefaultNetCoreTargetFramework)</TargetFramework>
-
-    <BuildHelixPayload>true</BuildHelixPayload>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/Grpc/test/InteropTests/InteropTests.csproj
+++ b/src/Grpc/test/InteropTests/InteropTests.csproj
@@ -4,8 +4,7 @@
     <ContainsFunctionalTestAssets>true</ContainsFunctionalTestAssets>
     <TargetFramework>$(DefaultNetCoreTargetFramework)</TargetFramework>
 
-    <!-- https://github.com/dotnet/aspnetcore/issues/24182 -->
-    <BuildHelixPayload>false</BuildHelixPayload>
+    <BuildHelixPayload>true</BuildHelixPayload>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/Grpc/test/testassets/InteropClient/InteropClient.cs
+++ b/src/Grpc/test/testassets/InteropClient/InteropClient.cs
@@ -96,7 +96,7 @@ namespace InteropTestsClient
             services.AddLogging(configure =>
             {
                 configure.SetMinimumLevel(LogLevel.Trace);
-                configure.AddSimpleConsole(loggerOptions =>
+                configure.AddConsole(loggerOptions =>
                 {
                     loggerOptions.IncludeScopes = true;
                 });

--- a/src/Grpc/test/testassets/InteropClient/InteropClient.cs
+++ b/src/Grpc/test/testassets/InteropClient/InteropClient.cs
@@ -34,6 +34,7 @@ using Grpc.Net.Client;
 using Grpc.Testing;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
 using Newtonsoft.Json.Linq;
 
 namespace InteropTestsClient
@@ -96,11 +97,11 @@ namespace InteropTestsClient
             var services = new ServiceCollection();
             services.AddLogging(configure =>
             {
-                configure.SetMinimumLevel(Microsoft.Extensions.Logging.LogLevel.Trace);
+                configure.SetMinimumLevel(LogLevel.Trace);
                 configure.AddSimpleConsole(loggerOptions =>
                 {
                     loggerOptions.IncludeScopes = true;
-                    loggerOptions.DisableColors = true;
+                    loggerOptions.ColorBehavior = LoggerColorBehavior.Disabled;
                 });
             });
 

--- a/src/Grpc/test/testassets/InteropClient/InteropClient.cs
+++ b/src/Grpc/test/testassets/InteropClient/InteropClient.cs
@@ -98,7 +98,9 @@ namespace InteropTestsClient
                 configure.SetMinimumLevel(LogLevel.Trace);
                 configure.AddConsole(loggerOptions =>
                 {
+#pragma warning disable CS0618 // Type or member is obsolete
                     loggerOptions.IncludeScopes = true;
+#pragma warning restore CS0618 // Type or member is obsolete
                 });
             });
 

--- a/src/Grpc/test/testassets/InteropClient/InteropClient.cs
+++ b/src/Grpc/test/testassets/InteropClient/InteropClient.cs
@@ -29,12 +29,10 @@ using Google.Apis.Auth.OAuth2;
 using Google.Protobuf;
 using Grpc.Auth;
 using Grpc.Core;
-using Grpc.Core.Utils;
 using Grpc.Net.Client;
 using Grpc.Testing;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Logging.Console;
 using Newtonsoft.Json.Linq;
 
 namespace InteropTestsClient
@@ -101,7 +99,6 @@ namespace InteropTestsClient
                 configure.AddSimpleConsole(loggerOptions =>
                 {
                     loggerOptions.IncludeScopes = true;
-                    loggerOptions.ColorBehavior = LoggerColorBehavior.Disabled;
                 });
             });
 

--- a/src/Grpc/test/testassets/InteropClient/Program.cs
+++ b/src/Grpc/test/testassets/InteropClient/Program.cs
@@ -17,6 +17,7 @@
 #endregion
 
 using System;
+using System.Reflection;
 
 namespace InteropTestsClient
 {
@@ -25,6 +26,9 @@ namespace InteropTestsClient
         public static void Main(string[] args)
         {
             Console.WriteLine("Application started.");
+
+            var runtimeVersion = typeof(object).GetTypeInfo().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "Unknown";
+            Console.WriteLine($"NetCoreAppVersion: {runtimeVersion}");
 
             InteropClient.Run(args);
         }

--- a/src/Grpc/test/testassets/InteropWebsite/Program.cs
+++ b/src/Grpc/test/testassets/InteropWebsite/Program.cs
@@ -23,6 +23,7 @@ using Microsoft.AspNetCore.Server.Kestrel.Core;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
 
 namespace InteropTestsWebsite
 {
@@ -37,7 +38,7 @@ namespace InteropTestsWebsite
             Host.CreateDefaultBuilder(args)
                 .ConfigureLogging(builder =>
                 {
-                    builder.AddSimpleConsole(o => o.DisableColors = true);
+                    builder.AddSimpleConsole(o => o.ColorBehavior = LoggerColorBehavior.Disabled);
                     builder.SetMinimumLevel(LogLevel.Trace);
                 })
                 .ConfigureWebHostDefaults(webBuilder =>

--- a/src/Grpc/test/testassets/InteropWebsite/Program.cs
+++ b/src/Grpc/test/testassets/InteropWebsite/Program.cs
@@ -37,7 +37,7 @@ namespace InteropTestsWebsite
             Host.CreateDefaultBuilder(args)
                 .ConfigureLogging(builder =>
                 {
-                    builder.AddSimpleConsole();
+                    builder.AddConsole();
                     builder.SetMinimumLevel(LogLevel.Trace);
                 })
                 .ConfigureWebHostDefaults(webBuilder =>

--- a/src/Grpc/test/testassets/InteropWebsite/Program.cs
+++ b/src/Grpc/test/testassets/InteropWebsite/Program.cs
@@ -23,7 +23,6 @@ using Microsoft.AspNetCore.Server.Kestrel.Core;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Logging.Console;
 
 namespace InteropTestsWebsite
 {
@@ -38,7 +37,7 @@ namespace InteropTestsWebsite
             Host.CreateDefaultBuilder(args)
                 .ConfigureLogging(builder =>
                 {
-                    builder.AddSimpleConsole(o => o.ColorBehavior = LoggerColorBehavior.Disabled);
+                    builder.AddSimpleConsole();
                     builder.SetMinimumLevel(LogLevel.Trace);
                 })
                 .ConfigureWebHostDefaults(webBuilder =>

--- a/src/Grpc/test/testassets/InteropWebsite/Startup.cs
+++ b/src/Grpc/test/testassets/InteropWebsite/Startup.cs
@@ -17,6 +17,7 @@
 #endregion
 
 using System;
+using System.Reflection;
 using Grpc.Testing;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.Extensions.DependencyInjection;
@@ -37,7 +38,13 @@ namespace InteropTestsWebsite
         public void Configure(IApplicationBuilder app, IHostApplicationLifetime applicationLifetime)
         {
             // Required to notify test infrastructure that it can begin tests
-            applicationLifetime.ApplicationStarted.Register(() => Console.WriteLine("Application started."));
+            applicationLifetime.ApplicationStarted.Register(() =>
+            {
+                Console.WriteLine("Application started.");
+
+                var runtimeVersion = typeof(object).GetTypeInfo().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "Unknown";
+                Console.WriteLine($"NetCoreAppVersion: {runtimeVersion}");
+            });
 
             app.UseRouting();
             app.UseEndpoints(endpoints =>

--- a/src/Shared/Process/ProcessEx.cs
+++ b/src/Shared/Process/ProcessEx.cs
@@ -218,8 +218,8 @@ namespace Microsoft.AspNetCore.Internal
         private static string GetNugetPackagesRestorePath() => (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("NUGET_RESTORE")))
             ? typeof(ProcessEx).Assembly
                 .GetCustomAttributes<AssemblyMetadataAttribute>()
-                .First(attribute => attribute.Key == "TestPackageRestorePath")
-                .Value
+                .FirstOrDefault(attribute => attribute.Key == "TestPackageRestorePath")
+                ?.Value
             : Environment.GetEnvironmentVariable("NUGET_RESTORE");
 
         public void Dispose()

